src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -571,10 +570,19 @@ private int StringValidation(int stringOffset)
                 for (var i = 0; i < escCount; i++)
                 {
                     var escCharOffset = ReadNumber(_mem + str + stringLength + escOffset + i, 1);
-                    var escChar = (char)ReadNumber(_mem + str + escCharOffset, 1);
-                    if (!(EscapeChars.Contains(escChar)))
-                        throw new InvalidDataException("String not valid");
-                }
+                    var escChar = ReadNumber(_mem + str + stringLength + escOffset - 1 - escCharOffset, 1);
+                    switch (escChar)
```

Comment:
Isn't this the reverse? If this is any of those, that is valid (also, add a comment what each number char is).
And only if it isn't one of those we need to throw